### PR TITLE
[Data Sources] Add: MemSQL query runner

### PR DESCRIPTION
**Pull request by alexanderlz**
https://github.com/memsql/memsql-python/

meanwhile the column types aren't available.
pending on memsql/memsql-python#8

### CODE REVIEW

File: redash/query_runner/memsql_ds.py [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+                },
+                "user": {
+                    "type": "string"
+                },
+                "password": {
+                    "type": "string"
+                }
+
+            },
+            "required": ["host", "port"]
```
Comment (arikfr) [inline redash/query_runner/memsql_ds.py]: Please add a `secret` field too (look at other data sources for example).

File: redash/query_runner/memsql_ds.py [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+                for table_name in filter(lambda a: len(a) > 0, map(lambda a: str(a['Tables_in_%s' % schema_name]),
+                                                                   self._run_query_internal(
+                                                                           tables_query % schema_name))):
+                    table_name = '.'.join((schema_name, table_name))
+                    columns = filter(lambda a: len(a) > 0, map(lambda a: str(a['Field']),
+                                                               self._run_query_internal(columns_query % table_name)))
+
+                    schema[table_name] = {'name': table_name, 'columns': columns}
+        except Exception, e:
+            raise sys.exc_info()[1], None, sys.exc_info()[2]
```
Comment (arikfr) [inline redash/query_runner/memsql_ds.py]: This is unnecessary.

File: redash/query_runner/memsql_ds.py [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+            # ====================================================================================================
+            columns = []
+            column_names = rows[0].keys() if rows else None
+
+            if column_names:
+                for column in column_names:
+                    columns.append({
+                        'name': column,
+                        'friendly_name': column,
+                        'type': None
```
Comment (arikfr) [inline redash/query_runner/memsql_ds.py]: It's better to use `TYPE_STRING` instead of `None`.

File: requirements.txt [hunk 4]
```diff
@@ -42,4 +42,6 @@ xlsxwriter==0.9.3
 pystache==0.5.4
 parsedatetime==2.1
 cryptography==1.4
+oauthlib==2.0.0
+WTForms==2.1
```
Comment (arikfr) [inline requirements.txt]: These two aren't needed.

File: redash/query_runner/memsql_ds.py (hunk 2 above)
Comment (alexanderlz) [inline redash/query_runner/memsql_ds.py]: the Exception handling?

Comment (arikfr) [inline redash/query_runner/memsql_ds.py]: Yes.

**Comment by arikfr** [comment]
Thanks. :)

**Pull request merged**